backend/app.py

```diff
@@ -1608,6 +1608,170 @@ def financial_assistant(subscriptionId):
+                "modification_time": datetime.now().isoformat(),
+            },
+        )
+
+        logging.info(f"Successfully removed Financial Assistant from subscription {subscriptionId}")
+
+        return create_success_response(
+            {
+                "message": "Financial Assistant removed from subscription successfully.",
+                "subscription": {
+                    "id": updated_subscription.id,
+                    "status": updated_subscription.status,
+                    "current_period_end": updated_subscription.current_period_end,
+                },
+            }
+        )
+
+    except stripe.error.InvalidRequestError as e:
+        logging.error(f"Stripe API error: {str(e)}")
+        return create_error_response(
+            "Invalid Subscription ID", HTTPStatus.NOT_FOUND
+        )
+    except stripe.error.StripeError as e:
+        logging.error(f"Stripe API error: {str(e)}")
+        return create_error_response(
+            "An error occurred while processing your request", HTTPStatus.BAD_REQUEST
+        )
+    except NotFound as e:
+        logging.warning(f"Not found: {str(e)}")
+        return create_error_response(str(e), HTTPStatus.NOT_FOUND)
+    except Exception as e:
+        logging.exception(f"Unexpected error: {str(e)}")
+        return create_error_response(
+            "An unexpected error occurred", HTTPStatus.INTERNAL_SERVER_ERROR
+        )
+
+#CHECK STATUS SUBSCRIPTION FA (FINANCIAL ASSITANT)
+@app.route("/api/subscription/<subscriptionId>/financialAssistant/status", methods=["GET"])
+@require_client_principal  # Security: Enforce authentication
+def get_financial_assistant_status(subscriptionId):
+    """
+    Check if Financial Assistant is added to a subscription.
+
+    Args:
+        subscriptionId (str): Unique Stripe Subscription ID
+
+    Returns:
+        JsonResponse: Response indicating if Financial Assistant is active in the subscription.
+        Success format:
+        {
+            "data": {
+                "financial_assistant_active": true,
+                "subscription": {
+                    "id": "<subscriptionId>",
+                    "status": "active"
+                }
+            }
+        }
+        
+    Raises:
+        NotFound: If the subscription is not found. HttpCode: 404
+        Unauthorized: If client principal ID is missing. HttpCode: 401
+    """
+    try:
+        # Consultar la suscripción en Stripe
+        subscription = stripe.Subscription.retrieve(subscriptionId)
+        
+        # Revisar si el Financial Assistant está activo en la suscripción
+        financial_assistant_active = any(
+            item.price.id == FINANCIAL_ASSISTANT_PRICE_ID for item in subscription["items"]["data"]
+        )
+        
+        # Responder con el estado del Financial Assistant
+        return jsonify({
+            "data": {
+                "financial_assistant_active": financial_assistant_active,
```

Comment:
@ramirezmorac2 return the financial assistant data
`subscription["items"]["data"]`